src/main/java/use_cases/PlayerGameInteractor.java

```diff
@@ -0,0 +1,212 @@
+        // turn it IS NOT a ComputerPlayer, as a User should not play as a ComputerPlayer
+        Player newCurrPlayer = currentGame.getCurrentTurn();
+        while((newCurrPlayer instanceof ComputerPlayer) & !(currentGame.hasWinner())) {
+            computerPlayerLogic((ComputerPlayer) newCurrPlayer);
+            newCurrPlayer = currentGame.getCurrentTurn();
+        }
+        // Return the current state of the game with a response model.
+        return new PlayerGameResponseModel(gameState);
+    }
+
+    /**
+     * Find the Player in the Game with the given name.
+     * This method assumes no Player in the same Game has the same name.
+     * @param name A name of a Player in the Game.
+     * @return The Player whose name matches name.
+     */
+    private Player findPlayerFromString(String name) {
+        for(Player p: currentGame.getPlayers()) {
+            if(p.getName().equals(name)) {
+                return p;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Find the Card in the given Player's Hand matching the description given.
+     * This method assumes the Card is in the Player's Hand (it wouldn't be called otherwise).
+     * @param value The value of the Card to return.
+     * @param suit The suit of the Card to return.
+     * @return The Card matching the given suit and value.
+     */
+    private Card findCardFromStrings(String value, String suit, Player player) {
+        for(Card card: player.getCards()) {
+            String valToCheck = card.getValue();
+            String suitToCheck = card.getSuit();
+            if(valToCheck.equals(value) & suitToCheck.equals(suit)) {
+                return card;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Implement the logic required when a Player tries to place the chosenCard down.
+     */
+    private void playCardRequestLogic(Card chosenCard, Player currentPlayer) {
+        boolean valid = currentGame.isValidCard(chosenCard);
+        if(valid) {
+            // The card is valid, place it down, change turns
+            currentPlayer.playCard(currentGame, chosenCard);
+            if(isWinner(currentPlayer)) {
+                winLogic(currentPlayer);
+            }
+            currentGame.changeCurrentTurn();
+            currentGame.notifyGameObservers();
+        }  // The card is not valid, therefore it should not be played, nothing is to be done.
+
+    }
+
+    /**
+     * Implement the logic required when a User requests to pick up a Card, that is, check if the user
+     * can pick up a Card, and if they can, then pick up one, otherwise don't do anything.
+     */
+    private void pickUpCardRequestLogic(Player currentPlayer) {
+        if(!(currentGame.getCurrentTurnHasPickedUp()) && !(anyValidCards(currentPlayer))) {
+            currentPlayer.pickUpCard(currentGame);
+            currentGame.notifyGameObservers();
+        }
+    }
```

Comment:
Are we going to set a limit on the max number of cards a user can pick up before their turn gets skipped?

Reply:
I noticed that for the computerPlayerLogic, after the computer player picks one card, their turn immediately gets skipped. Is this handled similarly for human players? Do you think this limit need to be enforced/implemented? Or perhaps I missed where that is in the current code.

Reply:
There is an attribute that Ive added to Game that checks whether or not the current Player has picked up a Card. If they have, they cannot pick up again, if it they have not (so it's false), then they can pick up if they dont have any valid cards in their deck.

Reply:
Got it, thanks!